src/Pact/Native.hs

```diff
@@ -1403,12 +1410,14 @@ base64decode = defRNative "base64-decode" go
   where
     go :: RNativeFun e
     go i as = case as of
-      [TLitString s] ->
-        case fromB64UrlUnpaddedText $ T.encodeUtf8 s of
-          Left e -> evalError' i
-            $ "Could not decode string: "
-            <> pretty e
-          Right t -> return $ tStr t
+      [TLitString s] -> do
+        base64Behavior <- (\legacy -> if legacy then Legacy else Simplified) <$> isExecutionFlagSet FlagDisablePact49
+        parseResult <- base64DecodeWithShimmedErrors (getInfo i) base64Behavior s
+        case parseResult of
+          Right bs -> case T.decodeUtf8' bs of
+            Right t -> return $ tStr t
+            Left _unicodeError -> evalError' i $ "invalid unicode"
+          Left base64Error -> evalError' i (pretty (T.pack base64Error))
```

Comment:
This is not aligned with line 1349:
```
...
Left e -> evalError' si (pretty e)
```

Reply:
Oh, you mean 1349 calls `pretty` on a `String` while 1422 calls `pretty` on a `Text`? That does seem like a discrepancy. I'll change both to call `pretty` on a `Text`, and recheck replay. Good catch!